From 8c83885bcaa7f94135aa33c5e4c2b177b3586b71 Mon Sep 17 00:00:00 2001
From: Xu Guangxin <guangxin.xu@intel.com>
Date: Mon, 7 Jun 2021 21:23:55 +0800
Subject: [PATCH 1/2] refenrece structure is ok

apply patch on top of 6669579959e881c0e18c6ace8e73476e70b9dbe5
./svc_encoder_rtc -sl 2 -lm 5 -b 4000 -bl 250,3750  1920x1080.y4m -o 1920x1080.ivf
---
 examples/svc_encoder_rtc.c | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/examples/svc_encoder_rtc.c b/examples/svc_encoder_rtc.c
index 7629a1b64..563b674e6 100644
--- a/examples/svc_encoder_rtc.c
+++ b/examples/svc_encoder_rtc.c
@@ -723,9 +723,11 @@ static void set_layer_pattern(
       layer_id->temporal_layer_id = 0;
       if (layer_id->spatial_layer_id == 0) {
         // Reference LAST, update LAST.
-        ref_frame_config->refresh[0] = 1;
+        for (int s = 0; s < 8; s++)
+          ref_frame_config->refresh[s] = 1;
         ref_frame_config->reference[SVC_LAST_FRAME] = 1;
       } else if (layer_id->spatial_layer_id == 1) {
+#if 0
         // Reference LAST and GOLDEN. Set buffer_idx for LAST to slot 1
         // and GOLDEN to slot 0. Update slot 1 (LAST).
         ref_frame_config->ref_idx[SVC_LAST_FRAME] = 1;
@@ -733,6 +735,8 @@ static void set_layer_pattern(
         ref_frame_config->refresh[1] = 1;
         ref_frame_config->reference[SVC_LAST_FRAME] = 1;
         ref_frame_config->reference[SVC_GOLDEN_FRAME] = 1;
+#endif
+        ref_frame_config->reference[SVC_LAST_FRAME] = 1;
       }
       break;
     case 6:
@@ -1271,10 +1275,13 @@ int main(int argc, const char **argv) {
   aom_codec_control(&codec, AV1E_SET_MV_COST_UPD_FREQ, 3);
   aom_codec_control(&codec, AV1E_SET_DV_COST_UPD_FREQ, 3);
   aom_codec_control(&codec, AV1E_SET_CDF_UPDATE_MODE, 1);
-  aom_codec_control(&codec, AV1E_SET_TILE_COLUMNS,
-                    cfg.g_threads ? get_msb(cfg.g_threads) : 0);
-  if (cfg.g_threads > 1) aom_codec_control(&codec, AV1E_SET_ROW_MT, 1);
-
+  aom_codec_control(&codec, AV1E_SET_SUPERBLOCK_SIZE, 0);
+  //aom_codec_control(&codec, AV1E_SET_TILE_COLUMNS,
+  //                  cfg.g_threads ? get_msb(cfg.g_threads) : 0);
+  //if (cfg.g_threads > 1) aom_codec_control(&codec, AV1E_SET_ROW_MT, 1);
+  aom_codec_control(&codec, AV1E_SET_TILE_COLUMNS,1);
+  aom_codec_control(&codec, AV1E_SET_TILE_ROWS,1);
+  aom_codec_control(&codec, AV1E_SET_NUM_TG, 4);
   svc_params.number_spatial_layers = ss_number_layers;
   svc_params.number_temporal_layers = ts_number_layers;
   for (i = 0; i < ss_number_layers * ts_number_layers; ++i) {
-- 
2.25.1

